src/explicit.rs

```diff
@@ -182,6 +190,22 @@ pub fn compute<'a, T: TextSource<'a> + ?Sized>(
             levels[i + j] = levels[i];
             processing_classes[i + j] = processing_classes[i];
         }
+
+        // Check if we need to start a new level run.
+        if i == 0 {
```

Comment:
if we're moving `level_runs` logic here then we should:

 - link to the spec
 - mention in the fn docs that this code also handles that part of the spec
 - ideally, more comments referencing spec list items or other text

Reply:
I've added/tidied up comments, as well as added a test for the edge-case of calling the paragraph API with empty text (and RTL direction, which prevents us taking the early exit in `compute_bidi_info_for_para`).